roles/kubernetes-apps/ingress_controller/ingress_nginx/defaults/main.yml

```diff
@@ -2,6 +2,8 @@
 ingress_nginx_namespace: "ingress-nginx"
 ingress_nginx_host_network: false
 ingress_nginx_service_type: LoadBalancer
+ingress_nginx_service_nodeport_http: {}
```

Comment:
why is it an empty dict?  what the value expected if filled ?

Reply:
@ant31 
Thank you for your feedback.
The intention is to leave the `.spec.ports[].nodePort` of the Service unspecified by default as it currently is(Dynamic port will be assigned).
If user specified `ingress_nginx_service_nodeport_http` or `ingress_nginx_service_nodeport_https` field in `addons.yml`, these values will be used as `.spec.ports[].nodePort`.

As a result of my reconsideration, I think `""` is appropriate instead of `{}` and fixed.
In addition, user may be confused how to use these param, so I've added commented out example to `addons.yml` like that.

https://github.com/mochizuki875/kubespray/blob/2b24f3f3a5cffbe8eb8eea0eea1eb4042e22422a/inventory/sample/group_vars/k8s_cluster/addons.yml#L103-L104

Reply:
thank you !